utils/build/docker/nodejs/nextjs/src/app/tag_value/[tag]/[status]/route.js

```diff
@@ -8,6 +8,42 @@ export async function GET (request, { params }) {
 
   rootSpan?.setTag('appsec.events.system_tests_appsec_event.value', params.tag)
 
-  return NextResponse.json('Value tagged')
+  // read querystring
+  const query = Object.fromEntries(new URL(request.url).searchParams)
+
+  // always read body for API sec
+  const body = await getBody(request)
+
+  let response
+
+  if (params?.tag?.startsWith?.('payload_in_response_body') && request.method === 'POST') {
+    response = { payload: body }
+  } else {
+    response = 'Value tagged'
+  }
+
+  return NextResponse.json(response, {
+    status: params.status || 200,
+    headers: new Headers(query)
+  })
 }
 export const OPTIONS = GET
+export const POST = GET
+
+// somehow this function works but only when formData() is called before json()
+// i don't want to know why
+async function getBody (request) {
+  try {
+    return Object.fromEntries(await request.formData())
+  } catch (err) {}
+
+  try {
+    return await request.json()
+  } catch (err) {}
+
+  try {
+    return await request.text()
+  } catch (err) {}
+
+  return null
```

Comment:
This function only works properly when formData() iscalled before .json(), i don't know why, please make sure we're not missing something here

Reply:
It works because `formData()` method doesn't try to read content, so the buffer is full when we use `request.json()` method.